### DISCUSSION TITLE
能否在使用域名解析的时候不发送sni？

### DISCUSSION THREAD

**Discussion by moranno** [XTLS/Xray-core#2303] [Ideas]
之前讨论过遇到本地某些IP访问特定的IP的时候，特别是甲骨文，经常会被运营商sni阻断（IP被禁3分钟）
能否在使用域名访问的时候不发送sni呢？
因为域名解析在本地就完成了，xray client拿到的也只是服务器的IP，为何还需要发送sni呢？

**Comment by RPRX** [comment]
https://github.com/XTLS/Xray-core/issues/1767 人在国外开机场就算了，~~还白嫖甲骨文也是一绝~~
https://github.com/XTLS/Xray-core/issues/1692 这个被你等到了，v1.8.3 删了 Legacy VMess
https://github.com/XTLS/Xray-core/issues/1998 这个也快被你等到了，VLESS ECDH(E) 不耗性能

**Reply by moranno** [reply]
所以这次我识趣的开了discuss，没有开issue😅😅😅，你说的对，我之前说的对新技术的喜爱说一个原因，但还有一个原因也是想做未来的技术储备；伊朗现在不是sni白名单，所以想到不发sni不就可以了🤔️？之前我遇到的阻断问题，再使用空sni后确实再没有被阻断过，但是对于用户来说，域名更flexible😅

**Comment by RPRX** [comment]
不行了实在有点搞笑，这位朋友吧，开着机场赚着钱，人在国外没风险，还啥都想白嫖，技术吧半斤八两，**所以经常会来这里白嫖“关于如何更好地白嫖甲骨文”的技术支持**，我是蚌埠住了。你既然赚到了钱，花点小钱去买别家服务器，花点小钱去买技术支持并不过分对不对？

**Reply by moranno** [reply]
我是愿意花钱请你解释下我一楼的疑问的？
请开个价和给个数字货币地址。

**Comment by chika0801** [comment]
https://github.com/EAimTY/tuic/releases 

建议看看这个，客户端配置中有禁止发送SNI的参数，你把这协议当隧道就行了。

这是一个客户端使用链式代理拿它（TUIC）当隧道的配置示例，供参考下。

**Reply by moranno** [reply]
感谢，不过你知道为啥要发送sni呢？

**Reply by chika0801** [reply]
我不懂。我猜 TLS 建立连接时需要客户端发目的 SNI， 服务端才能根据SNI验证，和SSL证书有关。至于QUIC传输为什么客户端可选不发，也能验证我也不懂具体的，你可试试问AI。

**Reply by chika0801** [reply]
> 之前讨论过遇到本地某些IP访问特定的IP的时候，特别是甲骨文，经常会被运营商sni阻断（IP被禁3分钟）

我看你问这现象，建议花钱买其它家VPS（多买些，少用大厂或virmach racknerd吧）可以避免，这现象当时在大陆随机地区随机人随机用不同协议都会触发，最近这1个月好像反馈的又人少了。3月开始最多。

他和发不发SNI我也没测试过，你可以试试用TUIC协议客户端不发SNI来实验下。但是TUIC协议有几例报告说端口被封的，它的ISS区也有一个ISS，我自己用倒没遇到。

**Reply by chika0801** [reply]
https://github.com/chika0801/Xray-examples/blob/main/warning.md#memo-4

这个收集，我觉得有理，QUIC类目前搞它，

> 那谁会成为靶子就很明显了，这也好理解，假如你是 GFW 的供应商，最后交差个 FQ 封锁率才百分之几的东西，不太合适吧。 肯定先找用的人多的下手，也就是机场喜欢用的那些什么 SS / VMess，什么 Trojan，针对研究，一封一片，效果拔群。

你是供应端也不会首先查QUIC类的这理，协议百花倒是好事
